src/components/events/EventNewSelector.js

```diff
@@ -0,0 +1,100 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Form from 'react-bootstrap/Form';
+import TranNewContainer from './event_new/tran/TranNewContainer';
+import PlanNewContainer from './event_new/plan/PlanNewContainer';
+import AccomNewContainer from './event_new/accom/AccomNewContainer';
+
+export class EventNewSelector extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      EventNewForm: (
+        <PlanNewContainer
+          begin_time={this.props.begin_time}
+          end_time={this.props.end_time}
+          trip_id={this.props.trip_id}
+          close={this.props.close}
+        />
+      )
+    };
+    this.onChange = this.onChange.bind(this);
+  }
+
+  /***************************** Core functions *****************************/
+
+  /**
+   * Services dropdown for user to select which event they would like to add.
+   * @param e the event containing the name of event to add.
+   */
+  onChange(e) {
+    switch (e.target.value) {
+      case 'plan':
+        this.setState({
+          EventNewForm: (
+            <PlanNewContainer
+              begin_time={this.props.begin_time}
+              end_time={this.props.end_time}
+              trip_id={this.props.trip_id}
+              close={this.props.close}
+            />
+          )
+        });
```

Comment:
This is quite a creative approach—I bet this took a while to figure out! Good for you making this work, I suspected it wouldn't be trivial.

Reply:
Thank you!